iotilecore/iotile/core/dev/semver.py

```diff
@@ -342,8 +375,10 @@ def FromString(cls, range_string):
             disjuncts = [[conj]]
 
         #Check for ^X.Y.Z
-        elif range_string[0] == '^':
+        elif range_string[0] in('^', '>='):
```

Comment:
I don't think the semantics of the `>=` operator are correct here.  It looks like it behaves identically to `^` but >= should not have an upper bound, whereas `^` should.

Reply:
I was basing this on the documented comment --
>         ^X.Y.Z - matches all versions with the same leading nonzero digit
>           greater than or equal the given range.

Reply:
The documented behavior is correct for the ^ operator.  It should do "semantic versioning" compatible matching, which is any version with the same major version number and minor+patch >= the given version.  

However `>=` is not the same operation.  `>=` should just match any version number greater than or equal to the specified version with no upper bound.  So, for example:

`^1.2.0` would match (1.2.0, 1.3.0, but not 2.0.0) whereas
`>=1.2.0` would match (1.2.0, 1.3.0, and 2.0.0)

Reply:
Ah, I see. It turns out I didn't actually need to keep this change in since I used the verisioning a bit differently, so I took it out since my code wasn't exercising the path / no use case for it yet.